lib/surface/formatter.ex

```diff
@@ -376,13 +376,22 @@ defmodule Surface.Formatter do
   """
   @spec format_string!(String.t(), list(option)) :: String.t()
   def format_string!(string, opts \\ []) do
+    trailing_whitespace =
+      case Regex.run(~r/\s+$/, string) do
+        [match] -> match
+        nil -> nil
+      end
+
     parsed =
       string
       |> String.trim()
       |> Surface.Compiler.Parser.parse!(translator: Surface.Formatter.NodeTranslator)
 
-    # Ensure the :indent option is set
-    opts = Keyword.put_new(opts, :indent, 0)
+    # Ensure the :indent and :trailing_newline_on_input options are set
+    opts =
+      opts
+      |> Keyword.put_new(:indent, 0)
+      |> Keyword.put_new(:trailing_newline_on_input, !is_nil(trailing_whitespace))
```

Comment:
Can we call this `:trailing_newline` for brevity? Also, I think we want `Keyword.put`, not `Keyword.put_new`.

I don't think we want them to be able to pass this option. I think it should be an internal-only opt.

```suggestion
     |> Keyword.put(:trailing_newline, !is_nil(trailing_whitespace))
```